nilearn/decoding/decoder.py

```diff
@@ -476,9 +474,16 @@ def fit(self, X, y, groups=None):
         else:
             n_problems = 1
 
+        # Check if the size of the mask image and the number of features allow
+        # to perform feature screening.
+        selector = check_feature_screening(self.screening_percentile,
+                                           self.mask_img_,
+                                           self.is_classification)
         # Return a suitable screening percentile according to the mask image
-        self.screening_percentile_ = _adjust_screening_percentile(
-            self.screening_percentile, self.mask_img_, verbose=self.verbose)
+        if hasattr(selector, 'percentile'):
+            self.screening_percentile_ = selector.percentile
+        else:
+            self.screening_percentile_ = self.screening_percentile
```

Comment:
I am still a little worried that a user might pass `None` for the screening_percentile in order to disable screening, even though it is not explicitly allowed in the docstring. could we change that to 
```suggestion
            self.screening_percentile_ = 100
```

or handle the case where `self.screening_percentile is None` in any way that doesn't cause `n_final_features = ...` to crash?

Reply:
Yeah the condition is missing an `elif`:

```diff
@@ -482,6 +482,8 @@ class _BaseDecoder(LinearModel, RegressorMixin, CacheMixin):
         # Return a suitable screening percentile according to the mask image
         if hasattr(selector, 'percentile'):
             self.screening_percentile_ = selector.percentile
+        elif self.screening_percentile is None:
+            self.screening_percentile_ = 100.0
         else:
             self.screening_percentile_ = self.screening_percentile

```

Reply:
I also added test case for `screening_percentile = None`  in commit 1b54e10de028160c45083aa6c7e07387e5fcb2c2

Reply:
awesome! let's just wait for the CI to finish and as far as I am concerned we can merge